src/renderer/canvas/edge/types/edge-straight.ts

```diff
@@ -17,6 +27,28 @@ export const drawStraightLine = <N extends INodeBase, E extends IEdgeBase>(
   context.beginPath();
   context.moveTo(sourcePoint.x, sourcePoint.y);
   context.lineTo(targetPoint.x, targetPoint.y);
+
+  const edgeLineStyleType = edge.getEdgeLineStyle().type;
+  switch (edgeLineStyleType) {
+    case EdgeLineStyleType.DASHED:
+      context.setLineDash(DEFAULT_DASHED_LINE_PATTERN);
+      break;
+    case EdgeLineStyleType.DOTTED:
+      context.setLineDash(DEFAULT_DOTTED_LINE_PATTERN);
+      break;
+    case EdgeLineStyleType.SOLID:
+      context.setLineDash(DEFAULT_SOLID_LINE_PATTERN);
+      break;
+    case EdgeLineStyleType.CUSTOM: {
+      const dashPattern: number[] = edge.getEdgeLineStyle().dashPattern;
+      context.setLineDash(dashPattern);
+      break;
+    }
+    default:
+      context.setLineDash(DEFAULT_SOLID_LINE_PATTERN);
+      break;
+  }
```

Comment:
This is repeated in all three types, but it is all the same code. Additionally, rendering should be super fast which means all unnecessary calls should be removed. For example, for each edge calling `getEdgeLineStyle` creates a new object just to figure out the correct style - when rendering has a min of 25 FPS, this will start to impact the process.

I would suggest removing `getEdgeLineStyle` - no need for it because you have the latest information in the `this.style` of the edge model.
I would suggest having new function on the edge model, something like `getLineDashPattern` that returns `number[] | null`. In case of solid or undefined lineStyle, it returns null. In case of any other, it returns a default one (for dashed or dotted) or a user-defined one (in case of custom). 

Then this code (and in all three edge types) would be simple as:

```
const lineDashPattern = edge.getLineDashPattern();
context.setLineDashPatern(lineDashPattern ?? []);
```

Reply:
Thanks for your feedback. I've made the following changes to the code as you suggested:

- I removed the getEdgeLineStyle() function, as you pointed out that it is unnecessary and creates a new object each time it is called.

- I added a new function called getLineDashPattern() to the edge model. This function returns a number[] or null, depending on whether the edge has a custom line style or not.

- I updated the code to use the getLineDashPattern() function instead of getEdgeLineStyle().

I believe these changes will make the code more efficient and improve the rendering performance.